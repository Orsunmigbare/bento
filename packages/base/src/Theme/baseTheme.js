import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
