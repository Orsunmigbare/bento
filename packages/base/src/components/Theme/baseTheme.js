export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
