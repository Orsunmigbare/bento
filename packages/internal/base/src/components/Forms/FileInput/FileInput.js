import React, { Component } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import hideVisually from "../../../Styles/hideVisually"
import {
  themify,
  color,
  font,
  spacing,
  targetSize,
  fontFamily,
  borderRadius,
  fontWeight,
  themeProps
} from "@staccx/theme"
import themePropTypes from "../../constants/themePropTypes"
const tinycolor = require("tinycolor2")

class FileInput extends Component {
  constructor(props, context) {
    super(props, context)

    this.state = {
      uploadedFile: null
    }

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(e) {
    const { value } = e.target

    if (value.length) {
      console.log(value)
      const fileName = value.split("\\").pop()
      console.log(fileName)
      this.setState({ uploadedFile: fileName })
    }

    this.props.onChange(e)
  }

  render() {
    const {
      children,
      id,
      disabled,
      className,
      variant,
      accept,
      multiple,
      onChange,
      ...restProps
    } = this.props

    return (
      <FileWrapper variant={variant}>
        <Input
          type="file"
          accept={accept}
          className={className}
          id={id}
          disabled={disabled}
          onChange={this.handleChange}
          {...restProps}
        />
        <Label htmlFor={id} variant={variant}>
          {this.state.uploadedFile ? this.state.uploadedFile : children}
        </Label>
      </FileWrapper>
    )
  }
}

FileInput.themeProps = {
  wrapper: {
    name: "fileinput_wrapper",
    description: "FileInput wrapper style",
    type: themePropTypes.style
  },
  input: {
    name: "fileinput_input",
    description: "Input style â€“ default is hidden! Style the label instead.",
    type: themePropTypes.style
  },
  label: {
    name: "fileinput_label",
    description: "Label style",
    type: themePropTypes.style
  }
}

const FileWrapper = styled.div`
  ${themify(FileInput.themeProps.wrapper)};
`

const Input = styled.input`
  ${hideVisually};
  ${themify(FileInput.themeProps.input)};
`

const Label = styled.label`
  cursor: pointer;
  background-color: ${color.primary};
  color: ${p =>
    tinycolor
      .mostReadable(color.primary()(p), ["#fff"], {
        includeFallbackColors: true,
        level: "AA",
        size: "large"
      })
      .toString()};
  border-width: 0;
  border-radius: ${borderRadius};
  padding: ${spacing.tiny} ${spacing.medium};
  cursor: pointer;
  min-height: ${targetSize.normal};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  line-height: 32px;
  font-weight: ${fontWeight.bold};
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  transition: background 0.2s ease;

  &:hover,
  &:focus {
    outline: none;
    background-color: ${p =>
      tinycolor(color.primary()(p))
        .darken(5)
        .toString()};
    color: ${p =>
      tinycolor
        .mostReadable(tinycolor(color.primary()(p)).darken(5), ["#fff"], {
          includeFallbackColors: true,
          level: "AA",
          size: "large"
        })
        .toString()};
  }

  ${themify(FileInput.themeProps.label)};
`

FileInput.propTypes = {
  accept: PropTypes.string,
  children: themeProps.children.isRequired,
  disabled: PropTypes.bool,
  id: PropTypes.string.isRequired,
  input: PropTypes.any,
  onChange: PropTypes.func,
  className: PropTypes.string,
  ignoreBase: PropTypes.func,
  variant: PropTypes.string
}

FileInput.defaultProps = {
  disabled: false,
  input: {},
  onChange: null,
  className: ""
}

export default FileInput
