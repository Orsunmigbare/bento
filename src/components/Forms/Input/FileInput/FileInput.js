import React, { Component } from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import Upload from "../../../Icons/Upload/Upload"
import {
  color,
  font,
  spacing,
  targetSize,
  fontFamily,
  borderRadius,
  fontWeight,
  applyVariants,
  commonPropTypes,
  themePropTypes,
  hideVisually
} from "../../../../theming"
import ThemeComponent from "../../../Theme/ThemeComponent"
const tinycolor = require("tinycolor2")

class FileInput extends Component {
  constructor(props, context) {
    super(props, context)
    this.fileInput = React.createRef() // To get the number of uploaded files from the input in case of multiple files
    this.state = {
      uploadedFile: null
    }

    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(e) {
    const { value } = e.target

    if (this.fileInput.current.files.length > 1) {
      this.setState({
        uploadedFile:
          this.fileInput.current.files.length +
          " " +
          this.props.multipleFilesLabel
      })
    } else if (value.length) {
      this.setState({
        uploadedFile: this.props.showFileName
          ? value.split("\\").pop()
          : "1 " + this.props.singleFileLabel
      })
    }
    this.props.onChange && this.props.onChange(e)
  }

  render() {
    const {
      children,
      id,
      disabled,
      className,
      variant,
      accept,
      multiple,
      onChange,
      ...restProps
    } = this.props

    return (
      <FileWrapper
        variant={variant}
        className={className}
        title={this.state.uploadedFile}
      >
        <Input
          type="file"
          ref={this.fileInput} // To get the number of uploaded files from the input in case of multiple files
          accept={accept}
          id={id}
          disabled={disabled}
          onChange={this.handleChange}
          multiple={multiple}
          variant={variant}
          {...restProps}
        />
        <Label htmlFor={id} variant={variant}>
          <Icon variant={variant} />
          <span>
            {this.state.uploadedFile ? this.state.uploadedFile : children}
          </span>
        </Label>
      </FileWrapper>
    )
  }
}

FileInput.themeProps = {
  wrapper: {
    name: "FileInput_Wrapper",
    description: "FileInput wrapper style",
    type: themePropTypes.style
  },
  input: {
    name: "FileInput_Input",
    description: "Input style â€“ default is hidden! Style the label instead.",
    type: themePropTypes.style
  },
  label: {
    name: "FileInput_Label",
    description: "Label style",
    type: themePropTypes.style
  },
  icon: {
    name: "FileInput_Icon_Style",
    description: "Icon style",
    type: themePropTypes.style
  },
  iconComponent: {
    name: "FileInput_Icon_Component",
    description: "Icon component",
    type: themePropTypes.component
  }
}

const FileWrapper = styled.div`
  ${applyVariants(FileInput.themeProps.wrapper)};
`

const Input = styled.input`
  ${hideVisually};
  ${applyVariants(FileInput.themeProps.input)};
`

const Label = styled.label`
  cursor: pointer;
  background-color: ${color.primary};
  color: ${p =>
    tinycolor
      .mostReadable(color.primary()(p), ["#fff"], {
        includeFallbackColors: true,
        level: "AA",
        size: "large"
      })
      .toString()};
  border-width: 0;
  border-radius: ${borderRadius};
  padding: ${spacing.tiny} ${spacing.medium};
  cursor: pointer;
  min-height: ${targetSize.normal};
  text-decoration: none;
  text-align: center;
  display: inline-block;
  font-size: ${font.base};
  font-family: ${fontFamily.body};
  line-height: 32px;
  font-weight: ${fontWeight.bold};
  white-space: nowrap;
  justify-content: center;
  align-items: center;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  transition: background 0.2s ease;

  > span {
    display: flex;
    align-items: center;
  }

  &:hover,
  &:focus {
    outline: none;
    background-color: ${p =>
      tinycolor(color.primary()(p))
        .darken(5)
        .toString()};
    color: ${p =>
      tinycolor
        .mostReadable(tinycolor(color.primary()(p)).darken(5), ["#fff"], {
          includeFallbackColors: true,
          level: "AA",
          size: "large"
        })
        .toString()};
  }

  ${applyVariants(FileInput.themeProps.label)};
`

const IconComponent = ({ ...props }) => (
  <ThemeComponent
    tagName={FileInput.themeProps.iconComponent}
    fallback={Upload}
    {...props}
  />
)

export const Icon = styled(IconComponent)`
  display: block;
  height: 12px;
  width: 12px;
  margin-right: 3px;
  ${applyVariants(FileInput.themeProps.icon)};
`

FileInput.propTypes = {
  accept: PropTypes.string,
  children: commonPropTypes.children.isRequired,
  disabled: PropTypes.bool,
  id: PropTypes.string.isRequired,
  input: PropTypes.any,
  onChange: PropTypes.func,
  className: PropTypes.string,
  ignoreBase: PropTypes.func,
  variant: PropTypes.string,
  multipleFilesLabel: PropTypes.string,
  singleFileLabel: PropTypes.string,
  showFileName: PropTypes.bool
}

FileInput.defaultProps = {
  disabled: false,
  input: {},
  onChange: null,
  className: "",
  multipleFilesLabel: "files",
  singleFileLabel: "file",
  showFileName: true
}

export default FileInput
