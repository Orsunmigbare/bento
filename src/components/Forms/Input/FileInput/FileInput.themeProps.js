import {
  themePropTypes,
  createVariantsFunctionFactory
} from "../../../../theming"

const themeProps = {
  wrapper: {
    name: "FileInput_Wrapper",
    description: "FileInput wrapper style",
    type: themePropTypes.style
  },
  input: {
    name: "FileInput_Input",
    description: "Input style â€“ default is hidden! Style the label instead.",
    type: themePropTypes.style
  },
  label: {
    name: "FileInput_Label",
    description: "Label style",
    type: themePropTypes.style
  },
  icon: {
    name: "FileInput_Icon_Style",
    description: "Icon style",
    type: themePropTypes.style
  },
  iconComponent: {
    name: "FileInput_Icon_Component",
    description: "Icon component",
    type: themePropTypes.component
  }
}

export default createVariantsFunctionFactory(themeProps)
