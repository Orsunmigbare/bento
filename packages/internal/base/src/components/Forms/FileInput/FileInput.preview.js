import React from "react"
import FileInput from "./FileInput"

const preview = {
  title: "FileInput",
  category: "Components/Forms",
  tags: ["visual"],
  component: FileInput,
  render: props => (
    <div>
      <FileInput
        group="signer"
        id="idjadsj"
        onChange={() => console.log("You changed it")}
        {...props}
      >
        Upload something
      </FileInput>
    </div>
  )
}

export default preview
