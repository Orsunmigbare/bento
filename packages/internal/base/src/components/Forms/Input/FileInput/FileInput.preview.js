import React from "react"
import FileInput from "./FileInput"
import Wrapper from "../../../Layout/Wrapper/Wrapper"
import Input from "../Input"
import CheckBox from "../../CheckBox/CheckBox"

const preview = {
  title: "FileInput",
  category: "Components/Forms/Input",
  tags: ["visual"],
  component: FileInput,
  render: props => (
    <div>
      <FileInput
        group="signer"
        id="idjadsj"
        onChange={() => console.log("You changed it")}
        {...props}
      >
        Upload something
      </FileInput>
    </div>
  ),
  renderExampleController: ({
    setComponentState,
    progress,
    multiple,
    showFileName
  }) => (
    <Wrapper>
      <div>
        <CheckBox
          id="1ui2bjqre"
          onChange={e => setComponentState({ multiple: !multiple })}
        >
          Multiple
        </CheckBox>
      </div>
      <div>
        <CheckBox
          id="asfdg"
          defaultChecked
          onChange={e => setComponentState({ showFileName: !!showFileName })}
        >
          Show file name
        </CheckBox>
      </div>
      <div>
        <Input
          label="Single file label"
          id="rtyukjhgfd"
          onChange={e => setComponentState({ singleFileLabel: e.target.value })}
        />
      </div>
      <div>
        <Input
          label="Multiple files label"
          id="567ikuyhjgfd"
          onChange={e =>
            setComponentState({ multipleFilesLabel: e.target.value })
          }
        />
      </div>
    </Wrapper>
  )
}

export default preview
