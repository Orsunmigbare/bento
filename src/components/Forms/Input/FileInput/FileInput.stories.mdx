import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import FileInput from "./FileInput"

<Meta title="Components|Forms/Input/FileInput" component={FileInput}/>


# FileInput


```javascript
import {FileInput} from "@staccx/bento"
```



## Standard FileInput

<Preview>
  <Story name="Standard">
   <FileInput id="9852145" >Last opp</FileInput>
  </Story>
</Preview>

## Disabled FileInput
Component takes a disabled prop to deactivate the button

<Preview>
  <Story name="Disabled">
   <FileInput id="98521ewf45" disabled={true}>Last opp</FileInput>
  </Story>
</Preview>



## Props for FileInput
<Props of={FileInput} />
