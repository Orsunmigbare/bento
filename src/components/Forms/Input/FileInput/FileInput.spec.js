import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../../theming/themes/baseTheme"
import FileInput from "./FileInput"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

const Child = () => <p>Body</p>

describe("FileInput", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <FileInput />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("with children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <FileInput>
              <Child />
            </FileInput>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("child and ID", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <FileInput id="test">
              <Child />
            </FileInput>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})

describe("Rendering", () => {
  it("Should render", () => {
    render(
      <ThemeProvider theme={baseTheme}>
        <FileInput id="test" data-testid="testID">
          <Child />
        </FileInput>
      </ThemeProvider>
    )
    expect(screen.getByTestId("testID")).toBeInTheDocument()
  })
})
