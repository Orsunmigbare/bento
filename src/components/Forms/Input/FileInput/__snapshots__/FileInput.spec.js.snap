// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput Snapshots No props 1`] = `
<div
  className="sc-AxirZ feQoIa"
  title={null}
>
  <input
    className="sc-AxiKw dVCywc"
    disabled={false}
    onChange={[Function]}
    type="file"
  />
  <label
    className="sc-AxhCb iAZYUs"
    disabled={false}
  >
    <svg
      className="sc-AxjAm bGsByD sc-AxhUy Vwrgh"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 .6l-.72.68-7 7 1.44 1.44L8 4.44V21h2V4.44l5.28 5.28 1.44-1.44-7-7L9 .6zM0 23v2h18v-2H0z"
      />
    </svg>
    <span />
  </label>
</div>
`;

exports[`FileInput Snapshots child and ID 1`] = `
<div
  className="sc-AxirZ feQoIa"
  title={null}
>
  <input
    className="sc-AxiKw dVCywc"
    disabled={false}
    id="test"
    onChange={[Function]}
    type="file"
  />
  <label
    className="sc-AxhCb iAZYUs"
    disabled={false}
    htmlFor="test"
  >
    <svg
      className="sc-AxjAm bGsByD sc-AxhUy Vwrgh"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 .6l-.72.68-7 7 1.44 1.44L8 4.44V21h2V4.44l5.28 5.28 1.44-1.44-7-7L9 .6zM0 23v2h18v-2H0z"
      />
    </svg>
    <span>
      <p>
        Body
      </p>
    </span>
  </label>
</div>
`;

exports[`FileInput Snapshots with children 1`] = `
<div
  className="sc-AxirZ feQoIa"
  title={null}
>
  <input
    className="sc-AxiKw dVCywc"
    disabled={false}
    onChange={[Function]}
    type="file"
  />
  <label
    className="sc-AxhCb iAZYUs"
    disabled={false}
  >
    <svg
      className="sc-AxjAm bGsByD sc-AxhUy Vwrgh"
      viewBox="0 0 24 24"
    >
      <path
        d="M9 .6l-.72.68-7 7 1.44 1.44L8 4.44V21h2V4.44l5.28 5.28 1.44-1.44-7-7L9 .6zM0 23v2h18v-2H0z"
      />
    </svg>
    <span>
      <p>
        Body
      </p>
    </span>
  </label>
</div>
`;
